### PR TITLE
Fix: set by default enable action if module is disabled instead of configure action

### LINKED ISSUE #35665: PrestaShop 8.1.4 - Problem with the "disabled" modules on the "Module Manager" page
### Discussed in https://github.com/PrestaShop/PrestaShop/discussions/35609

<div type='discussions-op-text'>

<sup>Originally posted by **Angar90** March 11, 2024</sup>
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Go to the "Module Manager" tab and disable one module.
After that the "Configure" button should be changed to "Enable", but Prestashop still displays the "Configure" button.

### Expected behavior

The "Configure" button should be changed to "Enable" like in the PrestaShop 1.7
Without this option is it hard to find which modules are disabled in the backoffice.

### Steps to reproduce

1.Go to the "Module Manager" tab.

2.Disable one module, for example the "Cross-selling" module:
https://prnt.sc/X4wVZC-uaLbq

3.After that the "Configure" text should be changed to "Enable" (in PrestaShop 1.7 it works correctly), but it still shows the text "Configure".

Screenshot from PrestaShop 8.1.4:
https://prnt.sc/xcrm_Bf--998

Screenshot from PrestaShop 1.7.8:
https://prnt.sc/QvuUMH3fKizV

Without this option is it hard to find which modules are disabled in the backoffice. I have this problem in all PrestaShop 8.x versions.

Also, sometimes I have a similar problem with the module's upgrade, after the upgrade, the old version number of the module is still displayed in the backoffice.

### PrestaShop version(s) where the bug happened

8.1.4

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_</div>

### PR DESCRIPTION
**Pull request by Nakahiru**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.2.x
| Description?      | [See description below](#description)
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See the issue https://github.com/PrestaShop/PrestaShop/issues/35665
| UI Tests          | https://github.com/Nakahiru/ga.tests.ui.pr/actions/runs/11213071238
| Fixed issue or discussion?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/35665
| Related PRs       | -
| Sponsor company   | -

## Description

This pull request fixes the issue at https://github.com/PrestaShop/PrestaShop/issues/35665. Since some PrestaShop versions, the action button (on the module list - module manager page) is not displaying the correct action. When a module is disabled, it shows the action `configure` instead of `enable` (in order to enable it again). This could potentially be misleading if a module is enabled or not.

### Before (without the pull request):
https://github.com/user-attachments/assets/0fd40f16-73f2-42d9-89d7-63271cd925dc

### After (with the pull request):
https://github.com/user-attachments/assets/b9b4cab4-2b55-4922-a867-c0f59675b6b7

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hi, thanks for this contribution!

I found some issues with the Pull Request description:

* The `branch` should be `develop` or `8.1.x`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))

Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!
### About linked issues
Please consider opening an issue before submitting a Pull Request:
* If it's a bug fix, it helps maintainers verify that the bug is effectively due to a defect in the code, and that it hasn't been fixed already.
* It can help trigger a discussion about the best implementation path before a single line of code is written.
* It may lead the Core Product team to mark that issue as a priority, further attracting the maintainers' attention.

(Note: this is an automated message, but answering it will reach a real human)

**Comment by boherm** [comment]
Hello!
Thanks for you PR, and yeah I think that could be better to do this fix in 8.1.x branch.
Can you change that? :)

**Comment by kpodemski** [comment]
Hello @Nakahiru 

Could you rebase onto 8.1.x branch? Alternatively, you could submit a new PR for this branch, so that everyone could benefit from this fix sooner :)

**Comment by Nakahiru** [comment]
Hey, thanks for your feedback. The pull request was rebased to 8.1.x 👍🏻

File: src/Adapter/Module/AdminModuleDataProvider.php [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <?php
+
```
Comment (boherm) [inline src/Adapter/Module/AdminModuleDataProvider.php]:
```suggestion
```

**Review by boherm — CHANGES_REQUESTED** [review]
Hey @Nakahiru!
Just a little comment about a extra line in your diff, but it's look good to me for the rest! :)

File: src/Adapter/Module/AdminModuleDataProvider.php (hunk 1 above)
Comment (Nakahiru) [reply src/Adapter/Module/AdminModuleDataProvider.php]: It was made automatically by the php lsp (intelephense). Removed 👍🏻

**Review by boherm — DISMISSED** [review]
LGTM :)

**Comment by Hlavtox** [comment]
Couldnt this be done just by moving the enable action to the top in the list, like it was before?

**Comment by Nakahiru** [comment]
🤔 I am not sure to understand. Is your proposal is to always set the enable action as the default (first) action without considering the context?

**Comment by Hlavtox** [comment]
@Nakahiru Yeah, that's how it worked before. When the module is enabled, there is no enable function.

I think if you do:
- enable
- Configure
- Disable
- Rest of them

it will work like you want to. :-)

**Comment by Nakahiru** [comment]
Thanks 🙏🏻 I didn't notice the protected array $moduleActions. Indeed, it's much simpler to put the enable action back at the top of this list instead of adding some additional logic 🙃
The change was made. It looks good.

**Review by kpodemski — APPROVED** [review]

**Review by Hlavtox — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @Nakahiru ,

Tested the actions buttons for disabling a module. Tested disabling a configurable/non-configurable module. Checked disabling multiple modules at the same time with Bulk actions.

https://github.com/user-attachments/assets/4909c480-9990-4a4d-b223-41931cbe26da

Tests autos : https://github.com/florine2623/testing_pr/actions/runs/11915077026 ✅ 

It is QA ✅

**Pull request merged**